### PR TITLE
Fixed documentation for transform_iterator

### PR DESCRIPTION
**Pull request by dparanic**

### CODE REVIEW

File: documentation/library_guide/extension_api.rst [hunk 1]
```diff
@@ -139,8 +139,14 @@ header.  All iterators are implemented in the ``oneapi::dpl`` namespace.
   The ``transform_iterator`` is an iterator defined over another iterator whose dereferenced value is the result
   of a function applied to the corresponding element of the original iterator.  Both the type of the original
   iterator and the unary function applied during dereference operations are required template parameters of
-  the ``transform_iterator`` class. The constructor of the ``transform_iterator`` receives both the original
-  iterator and an instance of the unary transform operation as well.
+  the ``transform_iterator`` class. There are three ways to call the constructor of the ``transform_iterator`` class:
```
Comment (timmiesmith) [inline documentation/library_guide/extension_api.rst:142]: This is good, but I think it's a little verbose with the repeated "Call it passing" and "in this way".  One alternative would be making the list the constructor interface, followed by the description.  Something like the following.

The ``transform_iterator`` class provides three constructors:
* ``transform_iterator()`` instantiates the iterator using default constructed base iterator and unary functor.
* ``transform_iterator(iter)`` instantiates the iterator using the base iterator provided and a default constructed unary functor.
* ``transform_iterator(iter, func)`` instantiates the iterator using the base iterator and unary functor provided.

I don't have a strong opinion on this though.  Either way is fine in my opinion.

Comment (dcbenito) [reply documentation/library_guide/extension_api.rst:142]: I like @timmiesmith suggestion and think that we should go with that.

**Review by dcbenito — CHANGES_REQUESTED** [review]
Please update with the comment from @timmiesmith.

File: documentation/library_guide/extension_api.rst (hunk 1 above)
Comment (dparanic) [reply documentation/library_guide/extension_api.rst]: I liked it too. Applied it.

Comment (dparanic) [reply documentation/library_guide/extension_api.rst]: I liked it too, applied.

**Review by ValentinaKats — APPROVED** [review]

**Pull request merged**